File: CS/DXApplication5/MVVM/Views/MainView.cs
using DevExpress.Utils.MVVM.Services;
using DevExpress.XtraEditors;
using DXApplication5.MVVM.Service;
using DXApplication5.MVVM.ViewModels;
using System;
using System.Collections.Generic;
using System.Linq;

namespace DXApplication5 {
    public partial class MainView : XtraForm {
        public MainView() {
            InitializeComponent();
            if (!mvvmContext.IsDesignMode)
                InitializeBindings();
        }

        void InitializeBindings() {
            mvvmContext.RegisterService(CustomDialogService.Create(this));
            var fluent = mvvmContext.OfType<DocumentsViewModel>();
            mvvmContext.RegisterDefaultService(DocumentManagerService.Create(navigationFrame));
            fluent.WithEvent(this, "Load")
                 .EventToCommand(x => x.LoadDefaultDocument());
        }
    }
}

File: VB/DXApplication5/MVVM/Views/MainView.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Utils.MVVM.Services
Imports DevExpress.XtraEditors
Imports DXApplication5.MVVM.Service
Imports DXApplication5.MVVM.ViewModels
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace DXApplication5
	Partial Public Class MainView
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
            If (Not mvvmContext.IsDesignMode) Then
                InitializeBindings()
            End If
		End Sub

		Private Sub InitializeBindings()
			mvvmContext.RegisterService(CustomDialogService.Create(Me))
			Dim fluent = mvvmContext.OfType(Of DocumentsViewModel)()
			mvvmContext.RegisterDefaultService(DocumentManagerService.Create(navigationFrame))
            fluent.WithEvent(Me, "Load").EventToCommand(Sub(x) x.LoadDefaultDocument())
        End Sub
    End Class
End Namespace

